apollo-router/src/protocols/multipart.rs

```diff
@@ -115,27 +116,33 @@ impl Stream for Multipart {
 
                     match self.mode {
                         ProtocolMode::Subscription => {
```

Comment:
This is where the actual change happens.
Now, we decide whether to send `payload` or `errors` in the response exclusively based on the `SUBSCRIPTION_ERROR_EXTENSION_KEY` being `true`.

The code is reordered a bit (special handling for graceful server-side close is moved earlier) so we aren't creating _another_ special-meaning internal message (`SubscriptionPayload` with `payload` being an empty object/null...)